### PR TITLE
fix: add p.source external check

### PR DESCRIPTION
**Pull request by kormanowsky**
#1365 + #1397 for v4

### CODE REVIEW

File: src/components/super/i-static-page/README.md [hunk 1]
```diff
@@ -260,6 +260,7 @@ const deps = {
  *   1. `lib` - the external library, i.e, something from `node_modules`
  *   2. `src` - the internal resource, i.e, something that builds from the `/src` folder
  *   3. `output` - the output library, i.e, something that builds to the `/dist/client` folder
+ *   4. `external` - externally hosted library, i.e. from a CDN
  *
  * @typedef {('lib'|'src'|'output')}
```
Comment (aabounegm) [inline src/components/super/i-static-page/README.md:265]: Shouldn't the new value be added to the type definition as well?

Comment (kormanowsky) [reply src/components/super/i-static-page/README.md]: Updated it, thanks!

File: src/components/super/i-static-page/README.md [hunk 2]
```diff
@@ -260,8 +260,9 @@ const deps = {
  *   1. `lib` - the external library, i.e, something from `node_modules`
  *   2. `src` - the internal resource, i.e, something that builds from the `/src` folder
  *   3. `output` - the output library, i.e, something that builds to the `/dist/client` folder
+ *   4. `external` - externally hosted library, i.e. from a CDN
  *
- * @typedef {('lib'|'src'|'output')}
+ * @typedef {('lib'|'src'|'output'|'external')}
```
Comment (aabounegm) [inline src/components/super/i-static-page/README.md:265]: I noticed now that it is `@typedef` not `@type`, which means it also needs a name:
```suggestion
 * @typedef {('lib'|'src'|'output'|'external')} LibSource
```

**Review by aabounegm — APPROVED** [review]
After writing the comment, I realized that it is in a Markdown file, so it doesn't really matter anyway)

**Review by aabounegm — APPROVED** [review]

**Review by shining-mind — APPROVED** [review]

**Pull request merged**